### PR TITLE
[IOPLT-133] Implements NumberPad component

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR creates the new NumberPad component

## List of changes proposed in this pull request
- Implements new `NumberPad` component

## How to test
Check related screen in example app

https://github.com/pagopa/io-app-design-system/assets/3959405/2a69bdb6-1158-49ab-a3c0-bbd39f907e85

### CODE REVIEW

File: src/components/numberpad/NumberPad.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import React, { ComponentProps, useRef } from "react";
+import { View } from "react-native";
+import { BiometricsValidType } from "../../utils/types";
+import { IONumberPadButtonStyles, IOStyles } from "../../core";
+import { VSpacer } from "../spacer";
+import { IconButton } from "../buttons";
+import { IOIconSizeScale, IOIcons } from "../icons";
+import { NumberButton } from "./NumberButton";
+
+type BiometricAuthProps =
```
Comment (drmarro) [inline src/components/numberpad/NumberPad.tsx:11]: can this type be just :
```
type BiometricAuthProps = {
  biometricType?: BiometricsValidType;
  onBiometricPress?: () => void;
  biometricAccessibilityLabel?: string;
};
```

File: src/components/numberpad/NumberPad.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  };
+
+  const onDeletePress = () => {
+    numberPadValue.current = numberPadValue.current.slice(0, -1);
+    onValueChange(numberPadValue.current);
+  };
+
+  type ButtonType = "biometric" | "delete";
+
+  const RowButtons = ({
```
Comment (drmarro) [inline src/components/numberpad/NumberPad.tsx:71]: Could this component be extracted as a standalone, reusable component, separate from the NumberPad?

Comment (CrisTofani) [reply src/components/numberpad/NumberPad.tsx:71]: Actually this should be only a local component avoiding code duplication to repeat button lines. Do you see any other possible usage of the component? 😄

File: src/components/numberpad/NumberPad.tsx (hunk 1 above)
Comment (CrisTofani) [reply src/components/numberpad/NumberPad.tsx:11]: Using the union type here defined we can handle the constraint of having all three props none of them defined at the same time: if we have a biometric option the tercet must be valorized correctly

**Comment by dmnplb** [comment]
As I can see from the simulator (and now from the attached video), the component does not seem to respect the experimental design mode.

**Review by dmnplb — CHANGES_REQUESTED** [review]
As I can see from the simulator (and now from the attached video), the component does not seem to respect the experimental design mode.

**Comment by CrisTofani** [comment]
> As I can see from the simulator (and now from the attached video), the component does not seem to respect the experimental design mode (in the disabled state)

fixed in [d684d67](https://github.com/pagopa/io-app-design-system/pull/127/commits/d684d678848fc1a8f8350871aad76d188e0b263f)

**Review by dmnplb — APPROVED** [review]

**Pull request merged**